optimum/habana/transformers/generation/utils.py

```diff
@@ -261,6 +248,7 @@ def _prepare_decoder_input_ids_for_generation(
         decoder_start_token_id: int = None,
         bos_token_id: int = None,
         device: torch.device = None,
+        max_new_tokens: int = None,
```

Comment:
Isn't it possible to get it from `self.generation_config.max_new_tokens` instead of passing it as an argument here?

Reply:
Hi, regisss, I think self.generation_config is from model hub like https://huggingface.co/meta-llama/Llama-2-7b-chat-hf/blob/main/generation_config.json. and generation_config is user defined config. if  user does not pass generation_config in . generation_config will be generated from self.generation_config. so do you think it's better to override parameter in self.generation_config using values from generation_config.

Reply:
You're right, let's keep the value from `generation_config`